### DISCUSSION TITLE
How to schedule and run CRON jobs?

### DISCUSSION THREAD

**Discussion by NikolaGrgic** [remix-run/remix#3833] [Q&A] [answered]
Hello,

I wanted to know if there was a recommended way of scheduling and running CRON jobs in Remix. I tried to import a file in the `entry.server.ts` file hoping to run the jobs, but it wouldn't trigger.

I know in Next.js, I can just create a custom server to run everything for me. Is the approach the same and recommended in Remix, or am I missing something?

Thank you.

**Comment by sergiodxa** [comment, marked as answer]
Remix itself doesn’t give you anything. As you said, you can create an Express server and run it there or you can use an external service.

**Reply by benheckmann** [reply]
Is `entry.server.ts` the correct place to start that scheduler? 

I.e. is something like this best practice:
```ts
// in entry.server.ts:
let serverInitialized = false;
let didInitialError = false;

async function ensureServerInitialized() {
  if (!serverInitialized && !didInitialError) {
    try {
      await startServer();
      serverInitialized = true;
      
      // Handle graceful shutdown
      const signals = ['SIGTERM', 'SIGINT'] as const;
      signals.forEach((signal) => {
        process.once(signal, async () => {
          logger.info(`Received ${signal}, shutting down...`);
          await stopServer();
          process.exit(0);
        });
      });
    } catch (error) {
      didInitialError = true;
      logger.error({ error }, 'Failed to initialize server');
      throw error;
    }
  }
}
```

**Comment by mehmettekn** [comment]
This is a good alternative. 
https://dev.to/remixtape/self-hosting-quirrel-5af7
